mail_activity_reminder/models/mail_activity.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2020 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from pytz import timezone, UTC
+
+from odoo import _, api, fields, models
+
+
+class MailActivity(models.Model):
+    _inherit = 'mail.activity'
+
+    last_reminder = fields.Date()
+
+    @api.model
+    def _get_activities_to_remind(self):
+        return self.search([]).filtered(lambda x: x._should_be_reminded())
```

Comment:
in a ctx whereas activities are used heavily, loading all activities every time could be an issue, no?
Maybe worth to add a TODO for future refactoring?

Reply:
Sorry, didn't get the issue. This code is called only via cron

Reply:
He is saying that a cron that loads all activities is too much. Can you restrict the search to mimic the `_should_be_reminded` method? Maybe it's worth to precompute in a stored computed field a field like `next_reminder_date` or similar.

Reply:
exactly. I bet this cron won't run once a day. Since it's going to load potentially thousands or records - if not more - we should find a way to limit the impact.
I don't want to block it because we can consider this and alpha stage but we should think about the improvement ASAP IMO.

Reply:
Taking into account that activities are removed when done (at least for now - in v14 there are plans to keep them), the scope is limited, but any way, crons should be as faster as possible.

Reply:
Refactored to rely on SQL filtering